scripts/data_overlap/compute_data_overlap_metrics.py

```diff
@@ -202,34 +151,25 @@ def compute_scenario_document_data_overlap(
 
     tokenizer: The tokenizer used to break the document into tokens
 
-    all_overlap_stats: The overlap stats for each scenario and n. The variable to write to.
+    stats_key_to_input_ids: Dict to keep track of input_ids that are overlapping
+
+    stats_key_to_reference_ids: Dict to keep track of reference_ids that are overlapping
 
-    ngram_counter: The ngrams that are overlapped between the training file and the scenario data
-    and their counts.
-    The outer dict maps from n to the inner dict, which maps from ngram to count.
     """
+
     document_tokens = tokenizer.tokenize(document)
     for n in ngram_index.keys():
         for document_ngram in ngrams(document_tokens, n):
             if document_ngram in ngram_index[n]:
                 for entry_overlap_key in ngram_index[n][document_ngram]:
-                    # update overlap_stats
-                    stats: DataOverlapStats = all_overlap_stats[entry_overlap_key.stats_key]
-                    stats.write_one_to_bit(entry_overlap_key.instance_id, entry_overlap_key.part)
-                    # skip the rest if max_overlapping_ngrams is 0
-                    if max_overlapping_ngrams != 0:
-                        if ngram_counter is None:
-                            raise ValueError("ngram_counter must be not none when max_overlapping_ngrams != 0")
-                        # update ngram_counter
-                        if entry_overlap_key not in ngram_counter:
-                            ngram_counter[entry_overlap_key] = {}
-                        if document_ngram in ngram_counter[entry_overlap_key]:
-                            ngram_counter[entry_overlap_key][document_ngram] += 1
-                        elif (
-                            max_overlapping_ngrams == -1
-                            or len(ngram_counter[entry_overlap_key]) < max_overlapping_ngrams
-                        ):
-                            ngram_counter[entry_overlap_key][document_ngram] = 1
+                    id = entry_overlap_key.instance_id
+                    part = entry_overlap_key.part
+                    if part == PART_INPUT:
+                        stats_key_to_input_ids[entry_overlap_key.stats_key].add(id)
+                    elif part == PART_REF:
+                        stats_key_to_reference_ids[entry_overlap_key.stats_key].add(id)
+                    else:
+                        hlog("Part neither input nor reference, hence not recording")
```

Comment:
This is going to be very noisy; remove this, make this debug only, or only log this every K times.